tests/tests_integration/test_api/test_time_series.py

```diff
@@ -112,6 +112,24 @@ def test_list_with_filters(self, cognite_client, post_spy):
         assert 0 == len(res)
         assert 1 == cognite_client.time_series._post.call_count
 
+    def test_list_timeseries_with_target_unit(self, cognite_client: CogniteClient) -> None:
+        ts1 = TimeSeries(external_id="test_list_timeseries_with_target_unit:1", unit_external_id="temperature:deg_c")
+        ts2 = TimeSeries(external_id="test_list_timeseries_with_target_unit:2", unit_external_id="temperature:deg_f")
```

Comment:
You need a random string at the end to avoid windows-ubuntu-runner-race-conditions